src/qonlinetranslator.cpp

```diff
@@ -269,6 +298,46 @@ bool QOnlineTranslator::isRunning() const
     return m_stateMachine->isRunning();
 }
 
+QString QOnlineTranslator::toJson() const
+{
+    QJsonObject translationOptions;
+    for (auto i = m_translationOptions.begin(); i != m_translationOptions.end(); ++i)
+    {
+        QJsonArray arr;
+        for (const auto &j : i.value())
+        {
+            arr.append(j.toJson());
+        }
+        translationOptions.insert(i.key(), arr);
+    }
+
+    QJsonObject examples;
+    for (auto i = m_examples.begin(); i != m_examples.end(); ++i)
+    {
+        QJsonArray arr;
+        for (const auto &j : i.value())
+        {
+            arr.append(j.toJson());
+        }
+        examples.insert(i.key(), arr);
+    }
```

Comment:
Some style changes (for consistency with used Qt coding style in this project):

* In C++ we usually use `it` for iterator variables, just for to distinguish them from indexes.
* In this project we open braces on the same line (except initializer-lists that you used for JSON objects).
* In this project we use `auto` only when type is obvious. There is even clang-tidy [check](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html) for it.

```suggestion
    QJsonObject translationOptions;
    for (auto it = m_translationOptions.begin(); it != m_translationOptions.end(); ++it) {
        QJsonArray options;
        for (const QOption &option : it.value()) {
            options.append(option.toJson());
        }
        translationOptions.insert(it.key(), options);
    }

    QJsonObject examples;
    for (auto it = m_examples.begin(); it != m_examples.end(); ++it) {
        QJsonArray examples;
        for (const QExample &example : it.value()) {
            examples.append(example.toJson());
        }
        examples.insert(it.key(), examples);
    }
```